applications/nrf_desktop/src/modules/hid_forward.c

```diff
@@ -86,10 +83,65 @@ static u8_t hidc_read(struct bt_gatt_hids_c *hids_c,
 		EVENT_SUBMIT(event);
 		usb_busy = true;
 	} else {
-		next_event = event;
+		next_mouse_event = event;
 	}
 
 	irq_unlock(key);
+}
+
+static void process_keyboard_report(const u8_t *data)
+{
+	struct hid_keyboard_event *event = new_hid_keyboard_event();
+
+	event->subscriber = usb_id;
+	event->modifier_bm = data[0];
+	memcpy(event->keys, &data[2], ARRAY_SIZE(event->keys));
+
+	u32_t key = irq_lock();
```

Comment:
please convert these irq_lock calls to spinlock

Reply:
done